tests/multi_ptr/multi_ptr_comparison_op.h

```diff
@@ -0,0 +1,586 @@
+            retrieved_results.first_first_mptr_result_val);
+    }
+    // If expected value isn't initialized, then this verification should be
+    // skipped
+    if (second_first_mptr_result_val) {
+      CHECK(second_first_mptr_result_val ==
+            retrieved_results.second_first_mptr_result_val);
+    }
+    // If expected value isn't initialized, then this verification should be
+    // skipped
+    if (first_second_mptr_result_val) {
+      CHECK(first_second_mptr_result_val ==
+            retrieved_results.first_second_mptr_result_val);
+    }
+  }
+};
+
+/**
+ * @brief Structure that combines variables used to validate test results for
+ *        operators that compares multi_ptr and nullptr
+ * @tparam T Data type that will be used in test
+ */
+template <typename T>
+struct mptr_nullptr_mptr_test_results {
+  // Expected value that will be returned after "nullptr *current operator*
+  // nullptr_mptr" operator will be called
+  bool nullptr_nullptr_mptr_result_val = true;
+  // Expected value that will be returned after "nullptr_mptr *current operator*
+  // nullptr" operator will be called
+  bool nullptr_mptr_nullptr_result_val = true;
+  // Expected value that will be returned after "nullptr *current operator*
+  // value_mptr" operator will be called
+  bool nullptr_value_mptr_result_val = true;
+  // Expected value that will be returned after "value_mptr *current operator*
+  // nullptr" operator will be called
+  bool value_mptr_nullptr_result_val = true;
+
+  /**
+   * @brief Verified test result. The object's data, that will call this
+   *        function, will used as expected results
+   * @param retrieved_results Retrieved test results
+   */
+  void verify_results(
+      const mptr_nullptr_mptr_test_results<T> &retrieved_results) const {
+    CHECK(nullptr_nullptr_mptr_result_val ==
+          retrieved_results.nullptr_nullptr_mptr_result_val);
+    CHECK(nullptr_mptr_nullptr_result_val ==
+          retrieved_results.nullptr_mptr_nullptr_result_val);
+    CHECK(nullptr_value_mptr_result_val ==
+          retrieved_results.nullptr_value_mptr_result_val);
+    CHECK(value_mptr_nullptr_result_val ==
+          retrieved_results.value_mptr_nullptr_result_val);
+  }
+};
+
+}  // namespace detail
+
+/**
+ * @brief Provides functor for verification on multi_ptr comparison operators
+ * @tparam T Current data type
+ * @tparam AddrSpaceT sycl::access::address_space enumeration's field
+ * @tparam IsDecoratedT sycl::access::decorated enumeration's field
+ */
+template <typename T, typename AddrSpaceT, typename IsDecoratedT>
+class run_multi_ptr_comparison_op_test {
+  static constexpr sycl::access::address_space space = AddrSpaceT::value;
+  static constexpr sycl::access::decorated decorated = IsDecoratedT::value;
+
+  using multi_ptr_t = sycl::multi_ptr<T, space, decorated>;
+
+  const T m_small_value = 1;
+  const T m_great_value = 2;
+  // Use an array to be sure that we have two elements that has subsequence
```

Comment:
```suggestion
  // Use an array to be sure that we have two elements that has consecutive
```

Reply:
I've applied this comment in this commit: https://github.com/KhronosGroup/SYCL-CTS/pull/363/commits/4bb647dcf71bb0462aa7edb7dcb81d59c29fb858